client/state/themes/actions/activate.js

```diff
@@ -37,13 +36,7 @@ export function activate(
 	keepCurrentHomepage = false
 ) {
 	return ( dispatch, getState ) => {
-		let showModalCondition =
-			! isJetpackSite( getState(), siteId ) && ! isSiteAtomic( getState(), siteId );
-
-		if ( isEnabled( 'themes/atomic-homepage-replace' ) ) {
-			showModalCondition =
-				! isJetpackSite( getState(), siteId ) || isSiteAtomic( getState(), siteId );
-		} else {
+		if ( ! isEnabled( 'themes/atomic-homepage-replace' ) ) {
 			// Keep default behaviour on Atomic. See https://github.com/Automattic/wp-calypso/pull/65846#issuecomment-1192650587
 			keepCurrentHomepage = isSiteAtomic( getState(), siteId ) ? true : keepCurrentHomepage;
 		}
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

Can we remove this condition and code as `themes/atomic-homepage-replace` is always true?

Reply:
I did, but reverted, because I think this was out of scope of this PR, which deals only with the modal, not the "persist/keep" homepage logic. We can do that in another PR and ask the relevant team for review, to be safe. Does this sound good? 😄

Reply:
That makes sense 👍